File: benchmarks/python/powellsumfcn.py
import numpy as np

def powellsumfcn(x: np.ndarray) -> np.ndarray:
    """
    Computes the value of the Powell Sum benchmark function.

    Parameters
    ----------
    x : numpy.ndarray
        Input array of size M-by-N.

    Returns
    -------
    numpy.ndarray
        Array of function values for each row of x. The shape of the array is M-by-1.

    Author: Mazhar Ansari Ardeh
    Please forward any comments or bug reports to mazhar.ansari.ardeh at Google's e-mail service or feel free to kindly modify the repository.
    """
    n = x.shape[1]
    absx = np.abs(x)

    scores = np.zeros((x.shape[0], 1))
    for i in range(n):
        scores += absx[:, i] ** (i + 1)

    return scores

File: src/matlab/powellsumfcn.m
% Computes the value of the Powell Sum benchmark function.
% SCORES = POWELLSUMFCN(X) computes the value of the Powell Sum function at
% point X. POWELLSUMFCN accepts a matrix of size M-by-N and returns a vetor
% SCORES of size M-by-1 in which each row contains the
% function value for the corresponding row of X.
% For more information, please visit:
% benchmarkfcns.info/doc/powellsumfcn
%
% Author: Mazhar Ansari Ardeh
% Please forward any comments or bug reports to mazhar.ansari.ardeh at
% Google's e-mail service or feel free to kindly modify the repository.
function scores = powellsumfcn(x)
    n = size(x, 2);
    absx = abs(x);

    % Create a vector of powers (1, 2, 3, ..., n+1)
    powers = 2:(n+1);

    % Compute the element-wise sum of (absx .^ powers)
    scores = sum(absx .^ powers, 2);
end
